test/extended/util/framework.go

```diff
@@ -726,216 +714,64 @@ var CheckImageStreamTagNotFoundFn = func(i *imageapi.ImageStream) bool {
 		strings.Contains(i.Annotations[imageapi.DockerImageRepositoryCheckAnnotation], "error")
 }
 
-// compareResourceControllerNames compares names of two resource controllers. It returns:
-//  -1 if rc a is older than b
-//   1 if rc a is newer than b
-//   0 if their names are the same
-func compareResourceControllerNames(a, b string) int {
-	var reDeploymentConfigName = regexp.MustCompile(`^(.*)-(\d+)$`)
-	am := reDeploymentConfigName.FindStringSubmatch(a)
-	bm := reDeploymentConfigName.FindStringSubmatch(b)
-
-	if len(am) == 0 || len(bm) == 0 {
-		switch {
-		case a < b:
-			return -1
-		case a > b:
-			return 1
-		default:
-			return 0
+func WaitForDeploymentConfig(kc kclientset.Interface, oc client.Interface, namespace, name string, version int64, cli *CLI) error {
+	start := time.Now()
+	err := wait.Poll(time.Second, 15*time.Minute, func() (done bool, err error) {
+		dc, err := oc.DeploymentConfigs(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
 		}
-	}
-
-	aname, averstr := am[0], am[1]
-	bname, bverstr := bm[0], bm[1]
-
-	aver, _ := strconv.Atoi(averstr)
-	bver, _ := strconv.Atoi(bverstr)
 
-	switch {
-	case aname < bname || (aname == bname && aver < bver):
-		return -1
-	case bname < aname || (bname == aname && bver < aver):
-		return 1
-	default:
-		return 0
-	}
-}
-
-// WaitForADeployment waits for a deployment to fulfill either isOK or isFailed.
-// When isOK returns true, WaitForADeployment returns nil, when isFailed returns
-// true, WaitForADeployment returns an error including the deployment status.
-// WaitForADeployment waits for at most a certain timeout (non-configurable).
-func WaitForADeployment(client kcoreclient.ReplicationControllerInterface, name string, isOK, isFailed func(*kapiv1.ReplicationController) bool, oc *CLI) error {
-	timeout := 15 * time.Minute
-
-	// closing done signals that any pending operation should be aborted.
-	done := make(chan struct{})
-	defer close(done)
-
-	// okOrFailed returns whether a replication controller matches either of
-	// the predicates isOK or isFailed, and the associated error in case of
-	// failure.
-	okOrFailed := func(rc *kapiv1.ReplicationController) (err error, matched bool) {
-		if isOK(rc) {
-			return nil, true
+		if dc.Status.LatestVersion > version {
+			return false, fmt.Errorf("deploymentconfig %s/%s latestVersion %d passed %d", namespace, name, dc.Status.LatestVersion, version)
 		}
-		if isFailed(rc) {
-			return fmt.Errorf("The deployment %q status is %q", name, rc.Annotations[deployapi.DeploymentStatusAnnotation]), true
+		if dc.Status.LatestVersion < version {
+			return false, nil
```

Comment:
Either include it in here or pass through an error that will be output instead of the error you get when you timeout. Likewise for the last `return false, nil` case where you should just print the status of the DC.